packages/cardano-services/src/Asset/DbSyncAssetProvider.ts

```diff
@@ -64,8 +65,18 @@ export class DbSyncAssetProvider extends DbSyncProvider implements AssetProvider
     const assetInfo: Asset.AssetInfo = { assetId, fingerprint, mintOrBurnCount, name, policyId, quantity };
 
     if (extraData?.history) await this.loadHistory(assetInfo);
-    if (extraData?.nftMetadata)
-      assetInfo.nftMetadata = await this.#dependencies.ntfMetadataService.getNftMetadata(assetInfo);
+    if (extraData?.nftMetadata) {
+      try {
+        // Use `null` instead of `undefined` when no nftMetadata is found / asset is not an NFT.
+        // Undefined means that there was a problem while querying and users can retry it.
+
+        // REVIEW: getNftMetadata does db-sync query or blockfrost query. Should we filter errros instead
+        // of catching all and transforming to undefined?
```

Comment:
@IvayloAndonov, can you help with some insight here?

Reply:
If the motivation of this PR is the same as for the [tokenMetadata](https://github.com/input-output-hk/cardano-js-sdk/pull/412), then we should handle the errors related to services that are not under our control (same as token registry service)
Since here we have two possible providers of nft metadata: `Blokfrost` is the external one, and `db-sync` is our concern I think here we could handle the HTTP 500 errors thrown by Blockfrost, then log and mark `nftMetadata` as undefined, if the behavior should be similar as for `tokenMetadata`

Not sure about `db-sync` though, there we have service discovery and re-resolving logic, so might be worth handling connection errors, set again `nftMetadata` as `undefined` and retry on the client I guess.
@mirceahasegan

Reply:
It is the same motivation as for [tokenMetadata](https://github.com/input-output-hk/cardano-js-sdk/pull/412).
Do we want to put the effort into filtering the Blockfrost errors, or can we go with the naive approach and catch all -> return undefined?

Reply:
Sorry, here in the context of `cardano-services` we perform only `db-sync` operations, blockfrost is completely separate provider, I was misled by the comment initially.
If the core types are not update then we shouldn't do anything with blockfrost
 @mirceahasegan